solc/jsonCompiler.cpp

```diff
@@ -223,7 +223,13 @@ string compile(StringMap const& _sources, bool _optimize, CStyleReadFileCallback
 				contractData["assembly"] = compiler.streamAssembly(unused, contractName, _sources, true);
 				output["contracts"][contractName] = contractData;
 			}
-
+		}
+		catch (...)
+		{
+			output["errors"].append("Unknown exception while generating JSON output.");
```

Comment:
Not sure if "JSON output" is the correct term here. This, for example, include the gas estimation.